cmd/collectors/rest/rest.go

```diff
@@ -337,12 +342,26 @@ func (r *Rest) PollData() (*matrix.Matrix, error) {
 			}
 		}
 
-		for key, metric := range r.Matrix.GetMetrics() {
-			f := instanceData.Get(key)
+		for _, metric := range r.prop.metrics {
+			f := instanceData.Get(metric.name)
 			if f.Exists() {
-				if err = metric.SetValueFloat64(instance, f.Float()); err != nil {
-					r.Logger.Error().Err(err).Str("key", key).Str("metric", metric.GetName()).
-						Msg("Unable to set float key on metric")
+				if metr, ok := r.Matrix.GetMetrics()[metric.name]; !ok {
+					if metr, _ = r.Matrix.NewMetricFloat64(metric.name); err != nil {
```

Comment:
would be slightly clearer to 1) create the metric by name and 2) update the map in this block and then remove the else and use it for both cases

Reply:
done